presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -423,9 +417,24 @@ else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ER
 
         validateSchema(partitionName, schema);
 
-        String fileNameWithExtension = fileName + getFileExtension(conf, outputStorageFormat);
+        String extension = getFileExtension(conf, outputStorageFormat);
+        String targetFileName;
+        if (bucketNumber.isPresent()) {
+            targetFileName = computeBucketedFileName(filePrefix, bucketNumber.getAsInt()) + extension;
+        }
+        else {
+            targetFileName = filePrefix + "_" + randomUUID() + extension;
+        }
+
+        String stagingFileName;
+        if (isWritingStagingFilesEnabled(session)) {
+            stagingFileName = ".tmp.presto." + filePrefix + "_" + randomUUID() + extension;
```

Comment:
Does Presto or any other system enumerate hidden files?

Reply:
@arhimondr : What do you mean by "enumerate hidden files?" -- does it matter ? :)

Reply:
It might matter if the `DIRECT_TO_TARGET_EXISTING_DIRECTORY` mode is used.

Reply:
@arhimondr: To my best knowledge I think Hive, Spark and Presto don't enumerate hidden files. But in general I don't think it matters because currently the engine might as well not clean up successfully when rolling back a failed transaction, in which case the file is not hidden. Also, the engine would commit staging files before registers in metastore, which is another layer of hidden. 

We could also add exception handling for file commit failure to delete all files with best effort.